.github/workflows/main.yml

```diff
@@ -87,8 +94,29 @@ jobs:
     - name: Setup and Run ppi_scaffolds tests
       run: |
         tar -xvf examples/ppi_scaffolds_subset.tar.gz -C examples
-        cd tests && uv run python test_diffusion.py
+        total_chunks=$(nproc)
+        cd tests 
+
+        #launch all chunks in background and record PIDs + labels
+        pids=""
+        for chunk_index in $(seq 1 $total_chunks); do
+          echo "Running chunk $chunk_index of $total_chunks"
+          uv run python test_diffusion.py --total_chunks $total_chunks --chunk_index $chunk_index &
+          pids="$pids $!"
+        done
+
+        # wait for each and track failures
+        fail=0
+        for pid in $pids; do
+          if ! wait "$pid"; then
+            echo "A chunk (PID $pid) failed"
+            fail=1
+          else
+            echo "A chunk (PID $pid) passed"
+          fi
+        done
```

Comment:
LGTM! I would personally would start to consider moving this logic into Python/shell script that live inside repository so it could be run/tested without the need to trigger the GH action. But for simple cases (like above) current approach  is also works. 👍